examples/PCovR_Regressors.ipynb

```diff
@@ -0,0 +1,325 @@
+   "metadata": {},
+   "source": [
+    "When there is no regressor supplied, PCovR uses `sklearn.linear_model.Ridge('alpha':1e-6, 'fit_intercept':False, 'tol':1e-12)`.\n",
+    "    \n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "9f1e2c99",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Regressor is Ridge(alpha=1e-06, fit_intercept=False, tol=1e-12) \n",
+      "\n",
+      "CPU times: user 3.72 ms, sys: 2.59 ms, total: 6.32 ms\n",
+      "Wall time: 2.98 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "pcovr1 = PCovR(\n",
+    "    mixing=mixing,\n",
+    "    n_components=2,\n",
+    ")\n",
+    "pcovr1.fit(X_scaled, y_scaled)\n",
+    "print(\"Regressor is\", pcovr1.regressor_, \"\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c3500b70",
+   "metadata": {},
+   "source": [
+    "# Use a fitted regressor"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a4c484d5",
+   "metadata": {},
+   "source": [
+    "You can pass a fitted regressor to PCovR to rely on the predetermined regression parameters. Currently, scikit-cosmo supports `scikit-learn` classes `LinearModel`, `Ridge`, and `RidgeCV`, with plans to support anu regressor with similar architecture in the future."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "10862180",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "regressor = Ridge(alpha=1e-6, fit_intercept=False, tol=1e-12).fit(X_scaled, y_scaled)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "be6b6088",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Regressor is Ridge(alpha=1e-06, fit_intercept=False, tol=1e-12) \n",
+      "\n",
+      "CPU times: user 2.48 ms, sys: 1.54 ms, total: 4.02 ms\n",
+      "Wall time: 1.52 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "pcovr2 = PCovR(\n",
+    "    mixing=mixing,\n",
+    "    n_components=2,\n",
+    "    regressor=regressor\n",
+    ")\n",
+    "pcovr2.fit(X_scaled, y_scaled)\n",
+    "print(\"Regressor is\", pcovr2.regressor_, \"\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6699c0d6",
+   "metadata": {},
+   "source": [
+    "# Use a pre-predicted y"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4bf200f4",
+   "metadata": {},
+   "source": [
+    "With `regressor='precomputed'`, you can pass a regression output $\\hat{Y}$ and optional regression weights $W$ to PCovR. If `W=None`, then PCovR will determine $W$ as the least-squares solution between $X$ and $\\hat{Y}$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "07b0d423",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "regressor = Ridge(alpha=1e-6, fit_intercept=False, tol=1e-12).fit(X_scaled, y_scaled)\n",
```

Comment:
Could you also add a `%%time` for this cell? I guess one does not gain much time wise with a pre-computed regressor, except when trying to tune the PCovR hypers (mixing & n_components).

Reply:
Done!